rules/no-invalid-remove-event-listener.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const methodSelector = require('./utils/method-selector');
+
+const MESSAGE_ID = 'no-invalid-remove-event-listener';
+const messages = {
+	[MESSAGE_ID]:
+		'Only call removeEventListener with the same reference to a function passed to addEventListener.'
+};
+
+const removeEventListenerSelector = methodSelector({
+	name: 'removeEventListener'
+});
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	return {
+		[removeEventListenerSelector]: node => {
+			const secondArgument = node.arguments[1];
+			if (!secondArgument) {
+				return;
+			}
+
+			if (
+				secondArgument.type === 'MemberExpression' ||
+				secondArgument.type === 'Identifier'
+			) {
```

Comment:
Instead of list allowed types, let's check the forbid types.

```js
foo.removeEventListener("scroll", getListener())
```

This should be valid.